lib/active_hash/relation.rb

```diff
@@ -143,15 +143,11 @@ def size
     end
 
     def pluck(*column_names)
-      symbolized_column_names = column_names.map(&:to_sym)
-
-      if symbolized_column_names.length == 1
-        column_name = symbolized_column_names.first
-        all.map { |record| record[column_name] }
+      if column_names.length == 1
+        all.map(&column_names.first.to_sym)
       else
-        all.map do |record|
-          symbolized_column_names.map { |column_name| record[column_name] }
-        end
+        # `tap with break` can be replaced with yield_self in Ruby 2.5 or then in Ruby 2.6
+        column_names.map { |column_name| all.map(&column_name.to_sym) }.tap { |values| break :zip.to_proc.(*values) }
```

Comment:
I think the bug you highlighted is we do not want to call `record[column_name]` but rather `record.public_send(column_name)`. (or `send`?)

This code has gone through at least 3 iterations:

```ruby
# first
column_names.map { |column_name| all.map(&column_name.to_sym) }.inject(&:zip)

#269
all.map do |record|
  symbolized_column_names.map { |column_name| record[column_name] }
end

#299
column_names.map { |column_name| all.map(&column_name.to_sym) }.then { :zip.to_proc.(*_1) }
```

I'm finding the `zip` and the `tap`/`then` a little complicated to mentally parse. (both versions of the zip code). Not sure what the extra traversals buy us.

Does something like this work for you?
(does send/public_send fix your bug?)
```ruby
all.map { |record| column_names.map { |column_name| record.public_send(column_name) } }
```

Reply:
Maybe the zip buys us some performance or something? Please share. Though, the non-zip version seems to be simpler for me.

Reply:
Thank you for your great feedback.
 I initially implemented pluck in a way that was closer to the original code.
However considering your feedback, proceeding as you've recommended seems better, especially from a performance standpoint.

Here are the results of the benchmark script:

```ruby
        Benchmark.bm 25 do |r|
          r.report "pluck(first)" do
            100000 times { CountryWithContinent.pluck_with_first(:id, :name, :continent) }
          end

          r.report "pluck(#269)" do
            100000 times { CountryWithContinent.pluck_with_269(:id, :name, :continent) }
          end

          r.report "pluck(#299)" do
            100000 times { CountryWithContinent.pluck(:id, :name, :continent) }
          end

          r.report "pluck(public_send)" do
            100000 times { CountryWithContinent.pluck_with_public_send(:id, :name, :continent) }
          end
        end
```

```
                                user     system      total        real
pluck(first)                2.755767   0.009342   2.765109 (  2.770909)
pluck(#269)                 0.962630   0.003945   0.966575 (  0.966671)
pluck(#299)                 2.175707   0.008026   2.183733 (  2.184002)
pluck(public_send)          0.764322   0.000021   0.764343 (  0.764447)

```

I'll move forward with the public_send approach. This seems like a step in the right direction. Thanks for your suggestion :smile:

Reply:
oh wow. thanks.
Yes, you were definitely following the original code's style. I appreciate that.
After looking at the numbers, I don't feel so bad wanting to go with what I consider a simpler style.